src/commands/infra/generate.ts

```diff
@@ -173,14 +170,78 @@ export const validateRemoteSource = async (
   return true;
 };
 
+/**
+ * Creates "generated" directory if it does not already exists
+ *
+ * @param projectPath Path to the definition.json file
+ */
+export const generateConfig = async (projectPath: string): Promise<void> => {
+  try {
+    // First, search for definition.json in current working directory
+    const templatePath = await parseDefinitionJson(projectPath);
+    const files = fs.readdirSync(".");
+    const cwdPath = process.cwd();
+    files.forEach(async file => {
+      // If there exists a definition.json, then read file
+      if (file === "definition.json") {
```

Comment:
@yradsmikham another example of the point I'm raising is that in this loop we are wholly dependent on there being a file system and a collection of files to iterate. When we find a file we read it via ` readDefinitionJson` and get an object back (`parentDefinitionJSON` or `leafDefinitionJSON`). From a unit test perspective we care about what is inside the defintionJSON _objects_.  

The meat of this loop is starts on line 194. That loop is our business logic that we can test without disk/network I/O.  Imagine if we had another method that started at line 194 and took the `parentDefinitionJSON` and `leafDefinitionJSON` objects  as method parameters and returned `parentDefinitionJSON.variables`

Just explaining the thought process. No changes needed

Reply:
Gotcha, that's a good point. I will make note of that and have it be an improvement for the next iteration. Thanks!

Reply:
I managed to resolve this in my latest commit!